src/Datasource/ConnectionManager.php

```diff
@@ -194,7 +194,7 @@ public static function dropAlias(string $name): void
      * @throws \Cake\Datasource\Exception\MissingDatasourceConfigException When config
      * data is missing.
      */
-    public static function get(string $name, bool $useAliases = true)
+    public static function get(string $name, bool $useAliases = true): ConnectionInterface
```

Comment:
This is because apparently connections can be literally anything.

Reply:
Addressed in https://github.com/cakephp/cakephp/pull/15863.